rust/emitter/src/opcode.rs

```diff
@@ -4,257 +4,257 @@ macro_rules! using_opcode_database {
+                (SetLocal, set_local, NULL, 4, 1, 1, JOF_LOCAL|JOF_NAME|JOF_DETECTING),
+                (SetAliasedVar, set_aliased_var, NULL, 5, 1, 1, JOF_ENVCOORD|JOF_NAME|JOF_PROPSET|JOF_DETECTING),
+                (SetIntrinsic, set_intrinsic, NULL, 5, 1, 1, JOF_ATOM|JOF_NAME|JOF_DETECTING),
+                (PushLexicalEnv, push_lexical_env, NULL, 5, 0, 0, JOF_SCOPE),
+                (PopLexicalEnv, pop_lexical_env, NULL, 1, 0, 0, JOF_BYTE),
+                (DebugLeaveLexicalEnv, debug_leave_lexical_env, NULL, 1, 0, 0, JOF_BYTE),
+                (RecreateLexicalEnv, recreate_lexical_env, NULL, 1, 0, 0, JOF_BYTE),
+                (FreshenLexicalEnv, freshen_lexical_env, NULL, 1, 0, 0, JOF_BYTE),
+                (PushVarEnv, push_var_env, NULL, 5, 0, 0, JOF_SCOPE),
+                (PopVarEnv, pop_var_env, NULL, 1, 0, 0, JOF_BYTE),
+                (EnterWith, enter_with, NULL, 5, 1, 0, JOF_SCOPE),
+                (LeaveWith, leave_with, NULL, 1, 0, 0, JOF_BYTE),
+                (BindVar, bind_var, NULL, 1, 0, 1, JOF_BYTE),
+                (DefVar, def_var, NULL, 5, 0, 0, JOF_ATOM),
+                (DefFun, def_fun, NULL, 1, 1, 0, JOF_BYTE),
+                (DefLet, def_let, NULL, 5, 0, 0, JOF_ATOM),
+                (DefConst, def_const, NULL, 5, 0, 0, JOF_ATOM),
+                (DelName, del_name, NULL, 5, 0, 1, JOF_ATOM|JOF_NAME|JOF_CHECKSLOPPY),
+                (Arguments, arguments, NULL, 1, 0, 1, JOF_BYTE),
+                (Rest, rest, NULL, 1, 0, 1, JOF_BYTE|JOF_TYPESET|JOF_IC),
+                (FunctionThis, function_this, NULL, 1, 0, 1, JOF_BYTE),
+                (Pop, pop, NULL, 1, 1, 0, JOF_BYTE),
+                (PopN, pop_n, NULL, 3, -1, 0, JOF_UINT16),
+                (Dup, dup, NULL, 1, 1, 2, JOF_BYTE),
+                (Dup2, dup2, NULL, 1, 2, 4, JOF_BYTE),
+                (DupAt, dup_at, NULL, 4, 0, 1, JOF_UINT24),
+                (Swap, swap, NULL, 1, 2, 2, JOF_BYTE),
+                (Pick, pick, NULL, 2, 0, 0, JOF_UINT8),
+                (Unpick, unpick, NULL, 2, 0, 0, JOF_UINT8),
+                (Nop, nop, NULL, 1, 0, 0, JOF_BYTE),
+                (Lineno, lineno, NULL, 5, 0, 0, JOF_UINT32),
+                (NopDestructuring, nop_destructuring, NULL, 1, 0, 0, JOF_BYTE),
+                (ForceInterpreter, force_interpreter, NULL, 1, 0, 0, JOF_BYTE),
+                (DebugCheckSelfHosted, debug_check_self_hosted, NULL, 1, 1, 1, JOF_BYTE),
+                (InstrumentationActive, instrumentation_active, NULL, 1, 0, 1, JOF_BYTE),
+                (InstrumentationCallback, instrumentation_callback, NULL, 1, 0, 1, JOF_BYTE),
+                (InstrumentationScriptId, instrumentation_script_id, NULL, 1, 0, 1, JOF_BYTE),
+                (Debugger, debugger, NULL, 1, 0, 0, JOF_BYTE),
             ]
         }
     }
 }
 
 macro_rules! define_opcode_enum {
-    ( [ $( ( $_op_upper:ident , $op_camel:ident , $( $_etc:tt )* ), )* ] ) => {
+    ( [ $( ( $op:ident , $( $_etc:tt )* ), )* ] ) => {
```

Comment:
I don't understand fully the set of changes to this to `op_upper` and `op_camel`. But, my guess is that we are no longer translating the enum into camel case / from uppercase? I think this (and the other two sections this appears) looks ok, but I am not familiar with this. If its critical might be worth having @arai-a or @nbp look at it

Reply:
This macro is deconstructing the content provided in the previous macro. 
```rust
using_opcode_database!(define_opcode_enum!());
```
calls the `using_opcode_database`, which in in charge of giving the argument to `define_opcode_enum` macro. The content is the above list, which got modified to exclude the `JSOP_*` syntax corresponding to the previous ignored `_op_upper`, and `op_camel` is replaced by `op`.

Note that Rust macro can matches lists of elements ` $( … )*`, where each element are tuples separated by a comma `( … ),` and where each element contains an identifier named `$op:ident`. Thus `$op` is expanded later on with `$( $op, )` which is the list of matched identifiers separated by commas.